src/api/goods.ts

```diff
@@ -1,19 +1,44 @@
 import { Good } from '../types/Good';
 
+function sortByName<T extends Good>(value: T[]): T[] {
+  return [...value].sort((prev, curr) => prev.name.localeCompare(curr.name));
+}
+
+function sliceByNumber(value: Good[], number: number): Good[] {
+  return [...value].slice(0, number);
+}
+
+function filterByColor(value: Good[], color: string): Good[] {
+  return [...value].filter(element => element.color === color);
```

Comment:
Methods slice and filter don't modify the array on which they are called, they return a new object, so the spread operator is redundant here